drainer/config_test.go

```diff
@@ -118,3 +121,39 @@ func (t *testDrainerSuite) TestAdjustConfig(c *C) {
 	c.Assert(cfg.ListenAddr, Equals, "http://0.0.0.0:8257")
 	c.Assert(cfg.AdvertiseAddr, Equals, "http://192.168.15.12:8257")
 }
+
+func (t *testDrainerSuite) TestConfigParsingFileWithInvalidOptions(c *C) {
+	yc := struct {
+		DataDir                string `toml:"data-dir" json:"data-dir"`
+		ListenAddr             string `toml:"addr" json:"addr"`
+		AdvertiseAddr          string `toml:"advertise-addr" json:"advertise-addr"`
+		UnrecognizedOptionTest bool   `toml:"unrecognized-option-test" json:"unrecognized-option-test"`
+	}{
+		"data.drainer",
+		"192.168.15.10:8257",
+		"192.168.15.10:8257",
+		true,
+	}
+
+	var buf bytes.Buffer
+	e := toml.NewEncoder(&buf)
```

Comment:
You may use a string instead, for example: https://github.com/pingcap/dm/pull/212/files#diff-fa28104e876d9e8e832e695718aa8edeR23

Reply:
I think struct is better for comparison when the demand changes so I did't change the code here.